cmd/soroban-cli/src/bin/doc-gen.rs

```diff
@@ -17,17 +17,86 @@ fn doc_gen() -> std::io::Result<()> {
         .show_table_of_contents(false)
         .title("Stellar CLI Manual".to_string());
 
-    let content = clap_markdown::help_markdown_custom::<soroban_cli::Root>(&options);
+    let content = generate_markdown_with_aliases::<soroban_cli::Root>(&options);
 
     std::fs::write(out_dir.join("FULL_HELP_DOCS.md"), content)?;
 
     Ok(())
 }
 
+fn generate_markdown_with_aliases<C: clap::CommandFactory>(
+    options: &clap_markdown::MarkdownOptions,
+) -> String {
+    let command = C::command();
+    let markdown_content = clap_markdown::help_markdown_custom::<C>(options);
+
+    // Split content into lines for better insertion efficiency
+    let mut lines: Vec<String> = markdown_content.lines().map(|s| s.to_string()).collect();
+
+    add_aliases_recursively(&command, &mut lines, 2);
+    
+    // Join the lines back into a single string after processing
+    lines.join("\n")
+}
+
+fn add_aliases_recursively(command: &clap::Command, lines: &mut Vec<String>, level: usize) {
+    for arg in command.get_arguments() {
+        let arg_name = format!("--{}", arg.get_id().as_str().to_kebab_case());
+
+        // Collect aliases and deduplicate
+        let mut alias_list = vec![];
+        let visible_aliases: Vec<String> = arg
+            .get_visible_aliases()
+            .into_iter()
+            .flatten()
+            .map(|alias| format!("--{}", alias.to_kebab_case()))
+            .collect();
```

Comment:
```suggestion
        for alias in arg
            .get_visible_aliases()
            .into_iter()
            .flatten()
            .map(|alias| format!("--{}", alias.to_kebab_case())) {
```
Why collect? and why is the arg_name added to the list of aliases?

Are are there commands with duplicate aliases?